### DISCUSSION TITLE
Best way to integrate additional javascript libraries

### DISCUSSION THREAD

**Discussion by jhudson10x** [helloflask/bootstrap-flask#250] [Q&A] [answered]
Say for example I want to use an autocomplete/typehead library like https://www.cssscript.com/autocomplete-typeahead-bootstrap-5/. What's going to be the best way to integrate this into a project using bootstrap-flask?

**Comment by PanderMusubi** [comment]
That site has some very useful scripts, thanks. 

I use often https://clipboardjs.com/ and https://www.chartjs.org/ Not specifically Bootstrap by the way.

**Comment by greyli** [comment, marked as answer]
Bootstrap-Flask has nothing to do with additional JS lib, you can include them directly:

```
... load Bootstrap JS ...
<script src="/path/to/autocomplete.js"></script>
```

If you use `bootstrap.load_js()` to include the Bootstrap JS lib in your template, load third-party lib after this call:

```
{{ bootstrap.load_js() }}
<script src="/path/to/autocomplete.js"></script>
```

**Comment by PanderMusubi** [comment]
@jhudson10x for your needs, see also https://github.com/wtforms/wtforms/issues/653

If you want you can upvote it or help with the implementation. 🍀
